plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java

```diff
@@ -1964,6 +1969,473 @@ public void testQueryTableFunction()
+        index(tableName, ImmutableMap.<String, Object>builder()
+                .put("id", 11)
+                .put("row1_t", ImmutableMap.<String, Object>builder()
+                        .put("f1", 12)
+                        .put("f2", 13)
+                        .put("row2_t", ImmutableMap.<String, Object>builder()
+                                .put("f1", 14)
+                                .put("f2", 15)
+                                .put("row3_t", ImmutableMap.<String, Object>builder()
+                                        .put("f1", 16)
+                                        .put("f2", 17)
+                                        .buildOrThrow())
+                                .buildOrThrow())
+                        .buildOrThrow())
+                .buildOrThrow());
+        index(tableName, ImmutableMap.<String, Object>builder()
+                .put("id", 21)
+                .put("row1_t", ImmutableMap.<String, Object>builder()
+                        .put("f1", 22)
+                        .put("f2", 23)
+                        .put("row2_t", ImmutableMap.<String, Object>builder()
+                                .put("f1", 24)
+                                .put("f2", 25)
+                                .put("row3_t", ImmutableMap.<String, Object>builder()
+                                        .put("f1", 26)
+                                        .put("f2", 27)
+                                        .buildOrThrow())
+                                .buildOrThrow())
+                        .buildOrThrow())
+                .buildOrThrow());
+
+        // Test select projected columns, with and without their parent column
+        assertThat(query("SELECT id, row1_t.row2_t.row3_t.f2 FROM " + tableName)).matches("VALUES (BIGINT '1', BIGINT '7'), (BIGINT '11', BIGINT '17'), (BIGINT '21', BIGINT '27')");
+        assertThat(query("SELECT id, row1_t.row2_t.row3_t.f2, CAST(row1_t AS JSON) FROM " + tableName))
+                .matches(
+                        "VALUES (BIGINT '1', BIGINT '7', JSON '%s'), "
+                                .formatted(
+                                        """
+                                                {
+                                                    "f1": 2,
+                                                    "f2": 3,
+                                                    "row2_t": {
+                                                        "f1": 4,
+                                                        "f2": 5,
+                                                        "row3_t": {
+                                                            "f1": 6,
+                                                            "f2": 7
+                                                        }
+                                                    }
+                                                }
+                                                """) +
```

Comment:
Formatting is off. Align the content of the text block with the opening `"""`. Apply the same to all the cases below.